tests/functional/s3/test_sync_command.py

```diff
@@ -164,3 +165,17 @@ def test_sync_with_delete_on_downloads(self):
         self.assertEqual(self.operations_called[0][0].name, 'ListObjects')
 
         self.assertFalse(os.path.exists(full_path))
+
+    def test_sync_skips_over_files_deleted_between_listing_and_transfer_attempt(self):
+        full_path = self.files.create_file('foo.txt', 'mycontent')
+        cmdline = '%s %s s3://bucket/' % (
+            self.prefix, self.files.rootdir)
+
+        # FileGenerator.list_files should skip over files that cause an OSError to be raised
+        # because they are missing when we try to get their stats.
+        with patch('awscli.customizations.s3.filegenerator.get_file_stat', side_effect=OSError()):
+            self.run_cmd(cmdline, expected_rc=0)
```

Comment:
Interesting. I think the ``expect_rc`` for this would be ``2`` as a warning gets thrown and the file gets skipped. And that is happening because I think you need to mock out the response for ``ListObjects`` by setting ``self.parsed_responses``. Take a look at the example here:
https://github.com/aws/aws-cli/pull/2333/files?diff=unified#diff-b8e88c17b1981210afdf4b277e015fb2R85